doc/source/dev/missing-bits.rst

```diff
@@ -48,6 +48,46 @@ additional parameters can be added to the function anywhere after the
 ``*``; new parameters do not have to be added after the existing parameters.
 
 
+Return Objects
+~~~~~~~~~~~~~~
+For new functions or methods that return two or more conceptually distinct
+elements, return the elements in an object type that is not iterable. In
+particular, do not return a ``tuple``, ``namedtuple``, or a "bunch" produced
+by `~scipy._lib._bunch.make_tuple_bunch`, the latter being reserved for adding
+new attributes to iterables returned by existing functions. Instead, use an
+existing return class (e.g. `~scipy.optimize.OptimizeResult`), a new, custom
+return class, or use `~dataclasses.make_dataclass`.
```

Comment:
I would not particularly advise `dataclasses.make_dataclass` as some IDE are not (yet) smart enough to correctly infer attributes then. Plain data classes is just perfect IMO.

Reply:
```suggestion
return class.
```